runner/utils/driver/istio/helpers.go

```diff
@@ -451,13 +451,60 @@ func (i *Istio) generateDeployment(service *model.Service, token string, listOfS
 	if strings.Contains(service.StatsInclusionPrefixes, "http.inbound") {
 		service.StatsInclusionPrefixes += ",http.inbound"
 	}
+
+	var nodeAffinity *v1.NodeAffinity
+	if len(service.Affinity) > 0 {
+		nodeAffinity = &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{},
+			},
+		}
+	}
+	podAffinity := &v1.PodAffinity{}
+	podAntiAffinity := &v1.PodAntiAffinity{}
+	for _, affinity := range service.Affinity {
+		switch affinity.Type {
+		case model.AffinityTypeService:
+			// affinity
+			if affinity.Weight > 0 {
+				required, preferred := getServiceAffinityObject(affinity)
+				if len(preferred) > 0 {
+					podAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(podAffinity.PreferredDuringSchedulingIgnoredDuringExecution, preferred...)
+				}
+				if len(required) > 0 {
+					podAffinity.RequiredDuringSchedulingIgnoredDuringExecution = append(podAffinity.RequiredDuringSchedulingIgnoredDuringExecution, required...)
+				}
+			} else {
+				affinityObj := affinity
+				affinityObj.Weight = -1 * affinityObj.Weight // make the weight positive
```

Comment:
Do this within `getServiceAffinityObject`. Pass an additional multiplier field which will be multiplied with the weight